### DISCUSSION TITLE
How to add the automatically created spans to already existing context...I have tried Span.storeInContext() and that seems to not work..

### DISCUSSION THREAD

**Discussion by GokulShrikanth** [open-telemetry/opentelemetry-java-instrumentation#3170] [Q&A] [answered]
It is possible to store the automatically created spans inside a context?...
`Span span = Span.current();
 Span.storeInContext(ParentContext);`

And this seems to be not working...How am I supposed to use this function when the span is created automatically?

**Comment by jkwatson** [comment, marked as answer]
The context is immutable, so you'll need to use the return value of `Span.storeInContext`. 

But, what are you trying to do with this code? It seems strange that you would store a span in the parent context, when it's already in the current context.

**Reply by GokulShrikanth** [reply]
Instead of the span being in the current context, I want it to be in the parentContext... The parentContext is from another service.. since I was not able to continue the trace....I am trying to have this span in the parentContext

**Reply by jkwatson** [reply]
I'm very confused by your statement. It sounds like something is going wrong with propagation and you're trying to work around it somehow?

**Reply by GokulShrikanth** [reply]
Yes, I am using Nats server to communicate between the services, unfortunately auto instrumentation doesn't support Nats...and the continuity is broken...I am trying to manually have them together.

**Reply by jkwatson** [reply]
Can you share some code? I suspect you just need to write the propagation code (ie getter) and hook it up to the nats request. The propagator will handle the context then.

**Reply by GokulShrikanth** [reply]
I am trying to solve this by partially using manual instrumentation...so from service A the context is sent, I am trying to make all the spans in service B to be a added on the same context as A

**Reply by GokulShrikanth** [reply]
```
TextMapSetter<JsonObject> setter  = new TextMapSetter<>() {
 @Override
  public void set(JsonObject carrier, String key, String value) {
    // Insert the context as Header
  carrier.put(key, value);
   }
  };

JsonObject context = new JsonObject();
 OpenTelemetry openTelemetry = GlobalOpenTelemetry.get();
 openTelemetry.getPropagators().getTextMapPropagator().inject(Context.current(), context, setter);
 JsonObject requestData = request.asJsonConnectorRequestObject().put("context",context);
 Logger.JOB_LOGGER.info("requestData " + requestData);`
```

Through NATS, I am able to send the context this way...and in the other service

```
JsonObject context= receivedBody.getJsonObject("context");
  Logger.PLATFORM_LOGGER.info(NATS| | context " + context);
  OpenTelemetry openTelemetry = GlobalOpenTelemetry.get();
  Tracer tracer = openTelemetry.getTracer("connector");
  
 TextMapGetter<JsonObject> getter = new TextMapGetter<>() {
  @Override
   public String get(JsonObject carrier, String key) {
   if (carrier.containsKey(key)) {
        return carrier.getValue(key).toString();
    }
        return null;
    }
  @Override
   public Iterable<String> keys(JsonObject carrier) {
   return carrier.fieldNames();
   }
};

 Context extractedContext = openTelemetry.getPropagators().getTextMapPropagator()
                                .extract(Context.current(), context, getter);
  Span span = tracer.spanBuilder("Connector Call").setParent(extractedContext).startSpan();
```

I am able to create this span under the context that I want...but the automatically generated ones aren't under this context...If there is a way to make the auto ones.. So is there a way to make the automatically generated ones be under this context?

**Reply by jkwatson** [reply]
I think you can just do


```
try (Scope scope = span.makeCurrent()) {
  // application code goes here
} finally {
  span.end();
}

```

**Reply by GokulShrikanth** [reply]
Thank you so much...This solved it
